src/keygen/Main.hs

```diff
@@ -12,19 +12,20 @@ import           System.FilePath      (takeDirectory)
 import           Universum
 
 import           Pos.Binary           (decodeFull, encode)
-import           Pos.Genesis          (GenesisData (..), getTotalStake)
-import           Pos.Types            (Coin, addressHash, makePubKeyAddress, mkCoin)
+import           Pos.Genesis          (GenesisData (..))
+import           Pos.Types            (addressHash, makePubKeyAddress)
 
-import           Avvm                 (genGenesis, getHolderId)
+import           Avvm                 (aeCoin, applyBlacklisted, genGenesis, getHolderId,
+                                       utxo)
 import           KeygenOptions        (AvvmStakeOptions (..), KeygenOptions (..),
                                        TestStakeOptions (..), optsInfo)
 import           Testnet              (genTestnetStakes, generateKeyfile)
 
 replace :: FilePath -> FilePath -> FilePath -> FilePath
 replace a b = toString . (T.replace `on` toText) a b . toText
 
-getTestnetGenesis :: Coin -> TestStakeOptions -> IO GenesisData
-getTestnetGenesis avvmStake tso@TestStakeOptions{..} = do
+getTestnetGenesis :: TestStakeOptions -> IO GenesisData
```

Comment:
You know that not considering avvm stake while taking N% of stake for richmen is formally incorrect =\